src/handle-completed.js

```diff
@@ -0,0 +1,15 @@
+const tasks = JSON.parse(localStorage.getItem('arrayOfTasks')) || [];
+
+export const handleComplected = (event, task) => {
+  if (!task.completed) {
+    task.completed = true;
+  } else {
+    task.completed = false;
+  }
+  return task;
+};
+
+export const removeCompetedTasks = (tasks) => {
+  const notCompletedTasks = tasks.filter((task) => !task.completed);
+  return notCompletedTasks;
+};
```

Comment:
:red_circle: [REQUIRED] I noticed that whenever you reload or create a new task, the changes made to the completed state are lost. This is because the local storage state is not being updated when the completed state is updated. Please make sure to update the local storage state whenever the completed state is updated.

![share 2](https://github.com/majdalkilany/To-Do-List/assets/26736582/ca97e30d-ed0c-4c47-9d89-5cef4679ca82)

Reply:
Hello @bienvenuushindi  thank you for your feedback 
I believe the array should be saved to the local storage after I click on clear all tasks,   kindly if you check the requirements and watch the provided video, and if you check the provided example website  you can not see it should not  be deleted on reload if it's marked as completed 

I think it should be saved after just when I click on the button  
  
![Screenshot 2023-06-16 155104](https://github.com/majdalkilany/To-Do-List/assets/60566840/4433cb1f-dfa8-4b0f-a1a9-6471367a591d)

If you check the screen shoot It mentions it  after the clear function

Reply:
Hi @majdalkilany 

Thanks for your feedback. I understand that you believe the array should only be saved to local storage after the "Clear All Tasks" button is clicked. However, I believe that the array should be saved to local storage whenever there is an action that changes the state of the array, such as updating the completed state, adding a new task, or removing a task. This is because the user should be able to reload the page and still see the changes that they have made or are making.

Reply:
@bienvenuushindi  
thank you for your feedback, but if you check the local storage it is saved to the local storage but there is no clear requirement to  render that with checked mark so that i did not implement it

Reply:
@majdalkilany 

Thanks for your comment. I understand that you are not sure what the project requirement means by "The updated array of items is stored in local storage."

The project requirement is asking you to save the array of items to local storage whenever the array is updated. This means that if a user marks a task as completed, the updated array should be saved to local storage. This way, when the user reloads the page, the updated array will be loaded and the user will see the correct list values.

There are a few reasons why a user might mark a task as completed and then clear it later. One reason is that they might want to keep an eye on their progress. By marking a task as completed, they can see how much progress they have made. They can also see which tasks they still need to complete.

Another reason why a user might mark a task as completed and then clear it later is that they might change their mind about whether or not the task is actually completed. For example, they might mark a task as completed but then realize that they actually need to do some more work on it. In this case, they can clear the task from the list and then add it back later when they are ready to work on it again.

I hope this clarifies the project requirement and the reasons why a user might mark a task as completed and then clear it later. 

Please let me know if you have any other questions.